config/CMakeLists.txt

```diff
@@ -67,14 +67,25 @@ if(HAVE_LANGINFO_H)
   check_symbol_exists(CODESET "langinfo.h" HAVE_NL_LANGINFO_CODESET)
 endif()
 
+check_include_files("endian.h" HAVE_ENDIAN_H)
+check_include_files("sys/endian.h" HAVE_SYS_ENDIAN_H)
+
+if(HAVE_ENDIAN_H)
+  set(ENDIAN_INCLUDE_FILE "endian.h")
+elseif(HAVE_SYS_ENDIAN_H)
+  set(ENDIAN_INCLUDE_FILE "sys/endian.h")
+else()
+  set(ENDIAN_INCLUDE_FILE "endian.h")
+endif()
```

Comment:
It would be more logical to write as

    set(ENDIAN_INCLUDE_FILE "endian.h")
    if(HAVE_SYS_ENDIAN_H)
      if(NOT HAVE_ENDIAN_H)
        set(ENDIAN_INCLUDE_FILE "sys/endian.h")
      endif()
    endif()

(maybe conditions may be joined like `HAVE_SYS_ENDIAN_H AND NOT HAVE_ENDIAN_H`, but I am not sure when this works and when not).